plutus-script-utils/src/Plutus/V1/Utils/Scripts.hs

```diff
@@ -0,0 +1,93 @@
+module Plutus.V1.Utils.Scripts
+    ( PV1.Datum
+    , PV1.DatumHash
+    , PV1.Redeemer
+    , PV1.RedeemerHash
+    , PV1.Validator
+    , PV1.ValidatorHash
+    , PV1.MintingPolicy
+    , PV1.MintingPolicyHash
+    , PV1.StakeValidator
+    , PV1.StakeValidatorHash
+    , datumHash
+    , redeemerHash
+    , validatorHash
+    , mintingPolicyHash
+    , stakeValidatorHash
+    , toCardanoApiScript
+    , scriptHash
+    , dataHash
+    , toCardanoAPIData
+    ) where
+
+import Cardano.Api qualified as Script
+import Cardano.Api.Shelley qualified as Script
+import Codec.Serialise (serialise)
+import Data.ByteString.Lazy qualified as BSL
+import Data.ByteString.Short qualified as SBS
+import Plutus.V1.Ledger.Scripts qualified as PV1
+import PlutusTx.Builtins qualified as Builtins
+
+-- | Hash a 'PV1.Datum builtin data.
+datumHash :: PV1.Datum -> PV1.DatumHash
+datumHash = PV1.DatumHash . dataHash . PV1.getDatum
+
+-- | Hash a 'PV1.Redeemer' builtin data.
+redeemerHash :: PV1.Redeemer -> PV1.RedeemerHash
+redeemerHash = PV1.RedeemerHash . dataHash . PV1.getRedeemer
+
+-- | Hash a 'PV1.Validator' script.
+validatorHash :: PV1.Validator -> PV1.ValidatorHash
+validatorHash =
+    PV1.ValidatorHash
+  . PV1.getScriptHash
+  . scriptHash
+  . PV1.getValidator
+
+-- | Hash a 'PV1.MintingPolicy' script.
+mintingPolicyHash :: PV1.MintingPolicy -> PV1.MintingPolicyHash
+mintingPolicyHash =
+    PV1.MintingPolicyHash
+  . PV1.getScriptHash
+  . scriptHash
+  . PV1.getMintingPolicy
+
+-- | Hash a 'PV1.StakeValidator' script.
+stakeValidatorHash :: PV1.StakeValidator -> PV1.StakeValidatorHash
+stakeValidatorHash =
+    PV1.StakeValidatorHash
+  . PV1.getScriptHash
+  . scriptHash
+  . PV1.getStakeValidator
+
+-- | Hash a 'Builtins.BuiltinData'
+dataHash :: Builtins.BuiltinData -> Builtins.BuiltinByteString
+dataHash =
+    Builtins.toBuiltin
+    . Script.serialiseToRawBytes
+    . Script.hashScriptData
+    . toCardanoAPIData
+
+-- | Convert a 'Builtins.BuiltinsData' value to a 'cardano-api' script
+--   data value.
+toCardanoAPIData :: Builtins.BuiltinData -> Script.ScriptData
+toCardanoAPIData = Script.fromPlutusData . Builtins.builtinDataToData
```

Comment:
Not sure if this used anywhere, but I would try to not export this function here. It exposes that `cardano-api` is used to hash things here and could lead to confusion. Users would likely reach for `cardano-api`'s and `plutus`' `fromPlutusData . fromBuiltin` anyway.